src/server/resultsets/ban_check.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#ifdef HAVE_MYSQLPP
+
+#include <ctime>
+
+#include "ban_check.hpp"
+
+ban_check::ban_check()
+{
+    ban_type = 0;
+    ban_duration = 0;
+    user_id = 0;
+    email = "";
+}
+
+void ban_check::read(mariadb::result_set_ref rslt)
+{
+    if(rslt->next())
+    {
+        ban_type = rslt->get_signed64("ban_type");
+        ban_duration = rslt->get_signed32("ban_end") != 0 ? rslt->get_signed32("ban_end") - std::time(nullptr) : 0;
+        user_id = rslt->get_signed32("ban_userid");
+        email = rslt->get_string("ban_email");
+    }
+    else
+    {
+        ban_duration = 1;
```

Comment:
This looks like a relict from the previous logic.

Reply:
Yep.